src/esphomelib/time/sntp_component.cpp

```diff
@@ -0,0 +1,59 @@
+//
+//  ntp_component.cpp
+//  esphomelib
+//
+//  Created by Brad Davidson on 06.06.18.
+//  Copyright © 2018 Otto Winter. All rights reserved.
+//
+
+#include <vector>
+
+#include "esphomelib/log.h"
+#include "esphomelib/helpers.h"
+#include "esphomelib/time/sntp_component.h"
+
+#ifdef USE_SNTP_COMPONENT
+
+ESPHOMELIB_NAMESPACE_BEGIN
+
+namespace time {
+
+static const char *TAG = "sntp";
+
+void SNTPComponent::setup(){
+  ESP_LOGCONFIG(TAG, "Setting up SNTP...");
+  global_sntp_component = this;
+}
+
+SNTPComponent::SNTPComponent() = default;
+
+void SNTPComponent::set_servers(const std::vector<std::string> &servers){
+  assert(servers.size() <= 3);
+
+  #ifdef ARDUINO_ARCH_ESP32
+  if(sntp_enabled()){
+    sntp_stop();
+  }
+  sntp_setoperatingmode(SNTP_OPMODE_POLL);
+  #endif
+  #ifdef ARDUINO_ARCH_ESP8266
+  sntp_stop();
+  #endif
+
+  for (uint8_t i = 0; i < servers.size(); i++){
+    sntp_setservername(i, strdup(servers[i].c_str()));
+  }
+  sntp_init();
```

Comment:
This initialization code should also go into the `setup()` method. The SNTP servers should be stored by three class members.

```cpp
std::string server_1_;
std::string server_2_;
std::string server_3_;
```

(Also, vectors can have a big overhead, because we know we will only have 3 max, we should declare these as separate variables)